docs/user/centralized_management.rst

```diff
@@ -0,0 +1,515 @@
+
+  ---------------------------------------------------------------------------
+  Host: host-2
+
+    [instance-id]                  [agent-class]           [state]   [target]
+    hm-host-2                      HostManager                  up        n/a
+    thermo3                        Lakeshore240Agent            up         up
+    aggregator                     AggregatorAgent              up         up
+
+
+The output is interpreted as follows.  After an initial statement of
+what site config file is being used, and the crossbar access address,
+a block is presented for each host in the SCF.  Within each host
+block, each agent instance-id is listed, along with its agent-class
+and values for "state" and "target".
+
+(See `below <Agents Running in Docker containers>`_ for an explanation
+of why the host-1-docker entries look odd.)
+
+``state`` and ``target``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``state`` column shows whether the Agent is currently running
+(``up``) or not (``down``).  This column may also show the value
+``unstable``, which indicates that an Agent keeps restarting (this
+usually indicates a code, configuration, or hardware error that is
+causing the agent to crash shortly after start-up).
+
+For the non-HostManager agents, the ``target`` column shows the state
+that HostManager will try to achieve for that Agent.  So if
+``target=up`` then the HostManager will start the Agent, and keep
+restarting the Agent if it crashes or otherwise terminates.  If
+``target=down`` then the HostManager will stop the Agent and
+not restart it.  (Note that in the case of Agents in docker
+containers, the HostManager will recognize if the Agent container has
+been started up and will proceed to stop it ``target=down``).
+
+Each HostManager can be commanded to change the target state of Agents
+it controls; see `Start/Stop Agents`_.
+
+For the HostManager lines, the ``target`` will always be ``[n/a]`` and
+the state will either be ``up``, ``down``, or ``sleeping``.  When the
+HostManager appears to be functioning normally, the state will be
+``up``.  If the HostManager appears to not be running at all, the
+state will be ``down``.  If the HostManager is running but the
+"manage" Process is not running for some reason, the state will be
+``sleeping``.
+
+Agents running in Docker containers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``ocsbow status`` output for Agent instances running in docker
+containers is a bit confusing.  In this example we seem to have two
+lines that refer to LSARR00::
+
+    LSARR00                        Lakeshore372Agent             ?          ?
+    ocs-LSARR00                    docker                       up         up
+
+The first line is there because the SCF listed an agent with class
+Lakeshore372Agent and instance-id LSARR00.  But because it was
+configured with ``'manage': 'docker'``, the HostManager is not
+managing that item directly.
```

Comment:
This looks as expected; you have a service for ocs-client but no entry in site-config.  The ? ? entry comes from having an entry in the site config that is marked as manage: docker, because those instances can't be _directly_ monitored by HostManager (rather, it can only monitor the docker containers that probably are hosting it).

Reply:
Right, I wasn't seeing a ? ? entry for the fake-data1 Agent, which has a `manage: docker` line. `ocs-client` isn't an Agent, just another container that I can run ocs clients from. Maybe some indexing was getting messed up by the ocs-client config existing? And the line should have read `fake-data1` for its instance-id instead?

I need to check this again with the latest version.

Reply:
Ah, sorry!

Looking into this, there is complex and buggy behavior when an instance-id and a service name are the same.  I'll address it.

I think the behavior should be that if you have an instance called X and it's marked with manage: 'docker', _and_ you also have a service named X, then ocs should assume those correspond and merge the configs in a helpful way.

Reply:
Ah, interesting. This might not be too common in practice, since we usually prefix "ocs-" to the service names. Sounds like a good edge case to handle though.